src/app/containers/reference-form-dialog/referenc-forms.config.ts

```diff
@@ -0,0 +1,148 @@
+  year: { type: 'number', label: 'Year' },
+  dateInCitation: { type: 'date', label: 'Date' },
+  articleTitle: { type: 'rich-text', label: 'Article title' },
+  source: { type: 'rich-text', label: 'Journal title' },
+  doi: { type: 'string', label: 'DOI' },
+  pmid: { type: 'string', label: 'PMID' },
+  extLink: { type: 'string', label: 'ext link' }
+};
+
+export const SOFTWARE_FORM: Record<string, FormControlConfigType> = {
+  year: { type: 'number', label: 'Year' },
+  dataTitle: { type: 'rich-text', label: 'Title' },
+  source: { type: 'rich-text', label: 'Source' },
+  publisherName: { type: 'string', label: 'Publisher Name' },
+  publisherLocation: { type: 'string', label: 'Publisher Location' },
+  version: { type: 'string', label: 'Version' },
+  doi: { type: 'string', label: 'DOI' },
+  pmid: { type: 'string', label: 'PMID' },
+  extLink: { type: 'string', label: 'ext link' }
+};
+
+export const CONFPROC_FORM: Record<string, FormControlConfigType> = {
+  year: { type: 'number', label: 'Year' },
+  articleTitle: { type: 'rich-text', label: 'Article title' },
+  conferenceName: { type: 'string', label: 'Conference name' },
+  conferenceLocation: { type: 'string', label: 'Conference location' },
+  conferenceDate: { type: 'date', label: 'Conference date' },
+  volume: { type: 'number', label: 'Volume' },
+  firstPage: { type: 'string', label: 'First page' },
+  lastPage: { type: 'string', label: 'Last page' },
+  elocationId: { type: 'string', label: 'eLocation ID' },
+  doi: { type: 'string', label: 'DOI' },
+  pmid: { type: 'string', label: 'PMID' },
+  extLink: { type: 'string', label: 'ext link' }
+};
+
+export const THESIS_FORM: Record<string, FormControlConfigType> = {
+  year: { type: 'number', label: 'Year' },
+  articleTitle: { type: 'rich-text', label: 'Article title' },
+  publisherName: { type: 'string', label: 'Publisher Name' },
+  publisherLocation: { type: 'string', label: 'Publisher Location' },
+  doi: { type: 'string', label: 'DOI' },
+  pmid: { type: 'string', label: 'PMID' },
+  extLink: { type: 'string', label: 'ext link' }
+};
+
+export const PATENT_FORM: Record<string, FormControlConfigType> = {
+  year: { type: 'number', label: 'Year' },
+  articleTitle: { type: 'rich-text', label: 'Article title' },
+  source: { type: 'rich-text', label: 'Source' },
+  publisherName: { type: 'string', label: 'Publisher Name' },
+  patent: { type: 'string', label: 'Patent' },
+  doi: { type: 'string', label: 'DOI' },
+  pmid: { type: 'string', label: 'PMID' },
+  extLink: { type: 'string', label: 'ext link' }
+};
+
+export function getFormConfigForType(type: ReferenceType): Record<string, FormControlConfigType> {
+  return {
+    journal: JOURNAL_FORM,
+    periodical: PERIODICAL_FORM,
+    book: BOOK_FORM,
+    report: REPORT_FORM,
+    data: DATA_FORM,
+    web: WEB_FORM,
+    preprint: PREPRINT_FORM,
+    software: SOFTWARE_FORM,
+    confproc: CONFPROC_FORM,
+    thesis: THESIS_FORM,
+    patent: PATENT_FORM
+  }[type];
```

Comment:
Wonder if it might be worth adding a bit of safety here to cope with a reference that doesn't have the type set, e.g.
```
return {}[type] || JOURNAL_FORM
```

Reply:
I intentionally make things bread cause having defaults may sometimes disguise bugs